catalog/app/containers/Admin/Sync.tsx

```diff
@@ -0,0 +1,306 @@
+import * as FF from 'final-form'
+import * as R from 'ramda'
+import * as React from 'react'
+import * as RF from 'react-final-form'
+import * as M from '@material-ui/core'
+
+import MetaTitle from 'utils/MetaTitle'
+import * as NamedRoutes from 'utils/NamedRoutes'
+import StyledLink from 'utils/StyledLink'
+import * as IPC from 'utils/electron/ipc-provider'
+import * as s3paths from 'utils/s3paths'
+import * as validators from 'utils/validators'
+
+import * as Form from './Form'
+import * as Table from './Table'
+
+type LocalFolderInputProps = M.TextFieldProps & Form.FieldProps
+
+export function LocalFolderInput({ input, ...props }: LocalFolderInputProps) {
+  const ipc = IPC.use()
+  const { onChange } = input
+
+  const handleClick = React.useCallback(async () => {
+    const newLocalPath = await ipc.invoke(IPC.EVENTS.LOCALPATH_REQUEST)
+    if (!newLocalPath) return
+    onChange(newLocalPath)
+  }, [ipc, onChange])
+
+  return (
+    <Form.Field
+      disabled={false}
```

Comment:
is this necessary?

Reply:
Fix https://github.com/quiltdata/quilt/pull/2829